ext/Pod-Html/lib/Pod/Html/Util.pm

```diff
@@ -30,75 +30,52 @@ Pod::Html::Util - helper functions for Pod-Html
 
 =head1 SUBROUTINES
 
-=head2 C<parse_command_line()>
+=head2 C<process_command_line()>
 
-TK
+Process command-line switches (options).  Returns a reference to a hash.  Will
+provide usage message if C<--help> switch is present or if parameters are
+invalid.
 
 =cut
 
-sub parse_command_line {
-    my $globals = shift;
-    my ($opt_backlink,$opt_cachedir,$opt_css,$opt_flush,$opt_header,
-        $opt_help,$opt_htmldir,$opt_htmlroot,$opt_index,$opt_infile,
-        $opt_outfile,$opt_poderrors,$opt_podpath,$opt_podroot,
-        $opt_quiet,$opt_recurse,$opt_title,$opt_verbose);
-
+sub process_command_line {
+    my %opts = map { $_ => undef } (qw|
+        backlink cachedir css flush
+        header help htmldir htmlroot
+        index infile outfile poderrors
+        podpath podroot quiet recurse
+        title verbose
+    |);
     unshift @ARGV, split ' ', $Config{pod2html} if $Config{pod2html};
-    my $result = GetOptions(
-                       'backlink!'  => \$opt_backlink,
-                       'cachedir=s' => \$opt_cachedir,
-                       'css=s'      => \$opt_css,
-                       'flush'      => \$opt_flush,
-                       'help'       => \$opt_help,
-                       'header!'    => \$opt_header,
-                       'htmldir=s'  => \$opt_htmldir,
-                       'htmlroot=s' => \$opt_htmlroot,
-                       'index!'     => \$opt_index,
-                       'infile=s'   => \$opt_infile,
-                       'outfile=s'  => \$opt_outfile,
-                       'poderrors!' => \$opt_poderrors,
-                       'podpath=s'  => \$opt_podpath,
-                       'podroot=s'  => \$opt_podroot,
-                       'quiet!'     => \$opt_quiet,
-                       'recurse!'   => \$opt_recurse,
-                       'title=s'    => \$opt_title,
-                       'verbose!'   => \$opt_verbose,
+    my $result = GetOptions(\%opts,
+        'backlink!',
```

Comment:
Minor nit, this can be greatly cleaned up by just throwing the option defs in a `qw//`; e.g.,:

```
$result = GetOptions(\%opts, qw/backlink! cachedir=s .../);
```